src/redefined.resolver.ts

```diff
@@ -23,11 +27,31 @@ export class RedefinedResolver implements Resolver {
         this.resolvers = options?.resolvers || RedefinedResolver.createDefaultResolvers();
     }
 
-    async resolve(domain: string, networks?: string[], options?: CustomResolverServiceOptions): Promise<Account[]> {
-        const customOptions = options || {};
-        return flatten(// @ts-ignore
-          await Promise.all(this.resolvers.map(it => it.resolve(domain, { throwErrorOnInvalidDomain: false, ...customOptions }, networks)))
-        ).filter(it => !networks || networks.includes(it.network) || it.network === "evm")
+    async resolve(domain: string, networks?: string[], options?: CustomResolverServiceOptions): Promise<ResolverResponse> {
+        const result = await Promise.allSettled(
+            this.resolvers.map(async resolver => {
+                try {
+                    return await resolver.resolve(domain, networks, options)
+                } catch (e: any) {
+                    throw Error(JSON.stringify({ vendor: resolver.vendor, error: e.message }))
```

Comment:
Instead of re-throwing new error we can just add it to data.errors. It will allow us to remove promise status filter below